applications/vanilla/modules/class.tagmodule.php

```diff
@@ -103,6 +103,13 @@ public function getData() {
 
         $this->autoContext();
 
+        $this->EventArguments['id'] = $this->ParentID;
+        $this->EventArguments['_TagData'] = &$this->_TagData;
```

Comment:
I would avoid attempting to give addons a backdoor to directly alter a protected property on the class. You can set a local tag data variable to a null value and pass that to the event. Addons hooking in can check to see if the value is still null before acting, so they don't overwrite each other's data. After all the handlers have been run through, `TagModule::getData` can check to see if addons populated the local variable and set its `_TagData` property as necessary.